storagegroup/storagegroup.go

```diff
@@ -19,11 +19,13 @@ import (
 // message. See ReadFromMessageV2 / WriteToMessageV2 methods.
 //
 // Instances should be created using one of the constructors.
-//
-// Note that direct typecast is not safe and may result in loss of compatibility:
-//
-//	_ = StorageGroup(storagegroup.StorageGroup) // not recommended
-type StorageGroup storagegroup.StorageGroup
+type StorageGroup struct {
```

Comment:
can you explain the motivation? we have it as a struct but we also have this: https://github.com/nspcc-dev/neofs-sdk-go/pull/598/commits/a07fde16f59c012e8e14a8900f11ba66adf47daf

can you at least explain it in commit? it does not look clear

Reply:
those are arrays, this is a struct

> can you at least explain it in commit?

done already

Reply:
> done already

it refs to another commit that says "This privatizes all fields." that is it?

> those are arrays, this is a struct

arrays also have some unique unwanted methods (slicing, element extraction by an index)

Reply:
> not sure i got ur point right

you said that the difference b/w `StorageGroup` and `user.ID` is structs vs arrays. but arrays can also be misused (they have indexes, etc). i don't mind either structs or just new types, just want to understand why sometimes it is done one way, but at a different time it is done another way (and commit message can help)

Reply:
> commit message can help

idk still what to write in it